### PR TITLE
[Do Not Merge]: Make StreamingFileSink extensible

### PR DESCRIPTION
**Pull request by kailashhd**
These changes will make the files extendable to add various metrics to the StreamingFileSink.

### CODE REVIEW

**Comment by kailashhd** [comment]
ptal @tweise @yxu-valleytider: Need to publish a jar like this temporarily till we get some information from the open source community on this.

File: flink-streaming-java/pom.xml [hunk 1]
```diff
@@ -41,21 +42,21 @@ under the License.
 		<dependency>
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-core</artifactId>
-			<version>${project.version}</version>
+			<version>${project.parent.version}</version>
```
Comment (tweise) [inline flink-streaming-java/pom.xml:45]: these pom changes are unnecessary

**Comment by tweise** [comment]
Which information is this waiting for? Is there an upstream JIRA and PR?

**Comment by kailashhd** [comment]
I started a discussion on the dev mailing list about this. I am waiting to hear some feedback on it and will file a JIRA for it. Will wait for a day before filing a JIRA in case I don't hear back about it. Will post the corresponding link here.

File: flink-streaming-java/pom.xml (hunk 1 above)
Comment (kailashhd) [reply flink-streaming-java/pom.xml:45]: Removing them. (I am not 100% sure on the release process of these artifacts). Will sync offline to understand more about it.

**Comment by yxu-valleytider** [comment]
OK with me for this approach.

**Pull request closed without merging**